lib/formdata.c

```diff
@@ -247,21 +261,16 @@ static CURLFORMcode FormAddCheck(struct FormInfo *first_form,
       /* our contenttype is missing */
       form->contenttype = strdup(type);
       if(!form->contenttype) {
-        return_value = CURL_FORMADD_MEMORY;
+        retval = CURL_FORMADD_MEMORY;
         break;
       }
       form->contenttype_alloc = TRUE;
     }
     if(form->name && form->namelength) {
       /* Name should not contain nul bytes. */
-      size_t i;
-      for(i = 0; i < form->namelength; i++)
-        if(!form->name[i]) {
-          return_value = CURL_FORMADD_NULL;
-          break;
-        }
-      if(return_value != CURL_FORMADD_OK)
-        break;
+      if(memchr(form->name, 0, form->namelength))
+        retval = CURL_FORMADD_NULL;
+      break;
```

Comment:
this is probably it

Reply:
sorry @jay, I totally missed your comment here as I fixed it up locally and force-pushed

Reply:
can you show me again what you think the problem was?

Reply:
oh I see it